### PR TITLE
meetup: improve descriptions by saying why each case is tested

### PR DESCRIPTION
**Pull request by petertseng**
descriptions show whether a date is the first, last, or an arbitrary
middle date of the week. This helps understand why certain cases are
selected.

Closes https://github.com/exercism/problem-specifications/issues/974

Please note the code that generated the descriptions as well.

### CODE REVIEW

**Comment by petertseng** [comment]
search for the string "another" (11 occurrences) where I was unable to discern a specific reason for a given test case. If you can figure out the reason, the descriptions for those can be improved.

Although analysing the new descriptions may indicate gaps in coverage, adding cases to fill those gaps is not for this PR to do.

**Review by ErikSchierboom — APPROVED** [review]
LGTM

Note: some tracks have issues with fairly long descriptions. I don't know if the current descriptions would be considered long for those tracks, but I wanted to note it.

File: exercises/meetup/canonical-data.json
Comment (petertseng) [inline exercises/meetup/canonical-data.json:1134]: I don't really know why this case is all the way down here. but I don't think moving it should be the job of this PR

**Comment by petertseng** [comment]
The longest description is "second Thursday in a month where it falls on another arbitrary day in the middle of the second week" at 99 characters. There are ways of shortening it; we'd just have to decide what information we're willing to sacrifice in the description, and/or what words can be safely dropped while losing no information.

for example, if we changed "in a month where" to "when", we'd have "second Thursday when it falls on another arbitrary day in the middle of the second week" and it'd save 12 characters.

Another solution is to group test cases together, for example all of the `second Tuesday` cases would be grouped together, the "second Tuesday" prefix removed from each of their descriptions, and the test runner would show the group in an appropriately-labeled way, such as

```
second Tuesday
  when it falls on the 8th, the first day of the second week - PASS
  when it falls on the 15th, the last day of the second week - PASS
  when it falls on an arbitrary day in the middle of the second week - PASS
```

that's an option that can be explored if needed, assuming we're comfortable relying on everyone doing the right thing with these descriptions that only make sense when combined with their group description.

**Comment by ErikSchierboom** [comment]
> for example, if we changed "in a month where" to "when", we'd have "second Thursday when it falls on another arbitrary day in the middle of the second week" and it'd save 12 characters.

This is definitely an option. Would changing "when it falls on" to "is" also be an option?

> Another solution is to group test cases together

It is, but only for tracks that support nested testS. I think the track that had issues with longer test names was the C track, which might or might not support nested tests. @wolf99 am I correct in remembering that the C or C++ track had issues with long test names?

**Comment by wolf99** [comment]
The C track does not "like" long test names. Technically they can be used but it results in quite ugly formatting.

For nested test cases, the C track usually concatenates the nesting name to avoid duplicate names.l across different nestings.

I am not sure about the C++ track.

**Comment by ErikSchierboom** [comment]
> The C track does not "like" long test names. Technically they can be used but it results in quite ugly formatting, which can get confusing for humans.

Do you know what the limit is for a long test name?

> For nested test cases, the C track usually concatenates the nesting name to avoid duplicate names across different nestings.

Ah, this is also what I do for C# and F#, so that doesn't really help with the length of the tests.

**Comment by petertseng** [comment]
Okay. So as a comparison:

    second Thursday in a month where it falls on another arbitrary day in the middle of the second week
    when second Thursday is another arbitrary day in the middle of the second week

that saves 21 characters.

**Comment by ErikSchierboom** [comment]
Yeah, that sounds fairly sensible too me. Could "is another arbitrary" be shortened to "is an arbitrary"?

Is that still too much @wolf99?

**Comment by wolf99** [comment]
This seems a good compromise (we don't want to get too convoluted purely to achieve terseness) 😊

**Comment by petertseng** [comment]
For cases with "another", I cannot change to "an" because there is already a case using "an"

**Review by ErikSchierboom — APPROVED** [review]
LGTM

**Comment by ErikSchierboom** [comment]
CC @exercism/reviewers

**Review by BethanyG — APPROVED** [review]
LGTM

**Comment by kotp** [comment]
> "is another arbitrary"
Can be changed to "any" for either or both of them.  "Arbitrary" and "any" are often interchangeable in meaning (just arbitrary makes one sound smarter?)

**Comment by petertseng** [comment]
thinking about it, you're right.

so for the test cases:

> when first Thursday is an arbitrary day in the middle of the first week
> when first Thursday is another arbitrary day in the middle of the first week

I would then change those two to:

> when first Thursday is some day in the middle of the first week
> when first Thursday is another day in the middle of the first week

thank you for the suggestion!

**Comment by petertseng** [comment]
EBWODP: Your suggestion has been applied and it saves 8 characters in the first case and 10 characters in the second! Great improvement!

**Review by junedev — APPROVED** [review]

**Comment by junedev** [comment]
I think the text length are very reasonable now. @petertseng Feel free to merge.

File: exercises/meetup/canonical-data.json
Comment (kotp) [inline exercises/meetup/canonical-data.json:1110]: Similarly, yes, it "happens to be" but it also simply "is"

Perhaps:

```suggestion
      "description": "last Wednesday in December that is also the last day of the year",
```

**Comment by petertseng** [comment]
I'd say it's a bit of a policy grey area whether I should be merging it at this time, because even though there are three approvals total, some changes came after some approvals.

In this case I'm using my judgment to say that overall the main changes have been looked by many, and the subsequent changes were small and all had at least two pairs of eyes on each (no one person has made them unilaterally and without review), and are not particularly likely to be contentious.

As you have seen before, with potentially contentious changes I would actually dismiss some prior approvals so that more approvals are needed.

However this is relying on a judgment call (what constitutes "potentially contentious"?), so if I turn out to be in error, let's discuss how to avoid future erroneous judgments in the future. For this one, I take responsibility for any problems that result.

**Comment by ErikSchierboom** [comment]
Thanks @petertseng!

**Pull request merged**
